javascript/ql/src/semmle/javascript/frameworks/Logging.qll

```diff
@@ -337,3 +337,49 @@ class StripAnsiStep extends TaintTracking::SharedTaintStep {
     )
   }
 }
+
+/**
+ * Provides classes and predicates for working with the `pino` library.
+ */
+private module Pino {
+  /**
+   * Gets a logger instance created by importing the `pino` library.
+   */
+  private API::Node pinoApi() {
+    result = API::moduleImport("pino").getReturn()
+    or
+    result = pinoApi().getMember("child").getReturn()
+  }
+
+  /**
+   * Gets a logger instance from the `pino` library.
+   */
+  private DataFlow::SourceNode pino() {
+    result = pinoApi().getAnImmediateUse()
+    or
+    // `pino` is installed as the "log" property on the request object in `Express` and similar libraries.
+    // in `Hapi` the property is "logger".
+    exists(HTTP::RequestExpr req |
+      result = req.flow().getALocalSource().getAPropertyRead(["log", "logger"])
```

Comment:
> I didn't like how I had to leave the API-graphs in the implementation.

I agree, this never feels nice.

Since request expressions should always have API nodes (they always come from external APIs), an alternative is simply to map backwards the `SourceNode` back to an API node (like we do in the Vuex PR). It feels a little dirty, but it's more forward compatible for when we port the HTTP model to API graphs.

Reply:
I went with the dirty "map back to the API node" approach.  

I'm looking forward to when we deprecate the `RequestExpr` class.